lib/sheets.ts

```diff
@@ -0,0 +1,175 @@
+            }, (err: any, result: any) => {
+                if (err) {
+                    // Handle error.
+                    reject(err);
+                } else {
+                    resolve(`${result.data} cells cleared.`);
+                }
+            });
+        }));
+    }
+
+    public readRow(row: number) {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.get({
+                range: this.name + "!" + row + ":" + row,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, res: any) => {
+                if (err) {
+                    reject('The API returned an error: ' + err);
+                }
+                const rows = res.data.values;
+                if (rows.length) {
+                    resolve(rows);
+                } else {
+                    reject('No data found.');
+                }
+            });
+        }));
+    }
+
+    public readCell(cell: string) {
+        return _await(new Promise((resolve, reject) => {
+            sheets.spreadsheets.values.get({
+                range: this.name + "!" + cell,
+                spreadsheetId: this.spreadsheetID,
+            }, (err: any, res: any) => {
+                if (err) {
+                    reject('The API returned an error: ' + err);
+                }
+                const rows = res.data.values;
+                if (rows.length) {
+                    resolve(rows);
+                } else {
+                    reject('No data found.');
+                }
+            });
+        }));
+    }
+
+    public appendRow(row: string[]) {
+        return _await(new Promise((resolve, reject) => {
+            const values = row;
+            const resource = {
+                values,
+            };
+            sheets.spreadsheets.values.append({
+                range: this.name,
+                resource,
+                spreadsheetId: this.spreadsheetID,
+                valueInputOption: 'RAW',
+            }, (err: any, result: any) => {
+                if (err) {
+                    // Handle error.
+                    reject(err);
+                } else {
+                    resolve(`${result.data.updates.updatedCells} cells appended.`);
+                }
+            });
+        }));
+    }
+}
+
+
+// Factory method to create spreadsheet by name, return spreadsheet class
+export function createSheet(name: string) {
+
+    let newID: string;
+
+    const resource = {
+        properties: {
+            title: name
+        },
+    };
+
+    sheets.spreadsheets.create({
+        fields: 'spreadsheetId',
+        resource,
+    }, (err: any, spreadsheet: any) => {
+        if (err) {
+            // Handle error.
+            log(err);
+        } else {
+            log(spreadsheet.data.spreadsheetId)
+            newID = spreadsheet.data.spreadsheetId;
+        }
+    });
+
+    return new Spreadsheet(newID, name);
```

Comment:
You need to use the Promise interface here. Wrap this callback ina  promise, and then `resolve(new Spreadsheet(newID, name)` with the new Spreadsheet. That way (1) your Spreadsheet object will get the correct ID (currently, it will just be undefined, since the `new Spreadsheet()` is run before the API callback can trigger, and (2) this function can't return until the Spreadsheet is _actually_ created, which will avoid us leaking reference to a spreadsheet that doesn't exist.